### PR TITLE
Discuss sources of variance in the user guide

### PR DESCRIPTION
**Pull request by matta**
It took me a long time to figure some of this stuff out, so it would be great to include it in the guide, assuming it is correct enough!

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -1266,12 +1269,54 @@ cpupower frequency-info -o proc
...
-good place to start looking for options.
+<a name="reducing-variance" />
+
+## Reducing Variance in Benchmarks
+
+The Linux CPU frequency governor [discussed
+above](#disabling-cpu-frequency-scaling) is not the only source of noise in
+benchmarks.  Some, but not all, of the sources of variance include:
+
+1. On multi-core machines not all CPUs run the same speed, so running a
```
Comment (LebedevRI) [inline docs/user_guide.md:1280]: Multi-socket?

Comment (matta) [reply docs/user_guide.md:1280]: Here I was thinking about things like TurboBoost, which is not limited to multi-socket.  Also, on my AMD Ryzen 9 5900X at least one CPU always seems to be running below the usual amount even though the performance governor is being used. Could this be thermal throttling? I'm not sure.

Then there are the various "boost" modes of Intel and AMD chips. Which reminds me, disabling that reduced variance on my system significantly, so I now mention it.

```sh
% cat /proc/cpuinfo | grep 'cpu MHz'                                                                                                                              /sys/devices/system/cpu/cpu0
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3442.683
cpu MHz		: 3700.000
cpu MHz		: 3402.804
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
cpu MHz		: 3700.000
```

Comment (LebedevRI) [reply docs/user_guide.md:1280]: Ok, then this should be something like `On multi-core machines not all CPUs/CPU cores/CPU threads run the same speed<...>`

**Comment by dominichamon** [comment]
could you pull this out of the user guide into its own document? i think it's important enough to get that but also isn't something someone who just wants to know how to use/run the thing needs to know right away.

File: docs/user_guide.md (hunk 1 above)
Comment (matta) [reply docs/user_guide.md]: Ok, went with that wording (though in the same commit I moved the content to a new document).

**Comment by matta** [comment]
> could you pull this out of the user guide into its own document? i think it's important enough to get that but also isn't something someone who just wants to know how to use/run the thing needs to know right away.

Done. I also moved the performance governor discussion to the new document, but a small forwarding paragraph in the user guide in case people have links there, or find the specific error message issued by the benchmark code there.

**Comment by matta** [comment]
Oopse, reopening.  I'm not very familiar with github and didn't realize what "close with comment" did.  ;-)

File: docs/user_guide.md (hunk 1 above)
Comment (LebedevRI) [reply docs/user_guide.md]: Thanks.

**Comment by dominichamon** [comment]
this is great. i think we'll also move the "random interleaving" into this reducing variance doc too as it is meant precisely for that.

let me know if you want to do that too or if not i'll just merge this.

**Review by dominichamon — APPROVED** [review]

**Comment by matta** [comment]
> let me know if you want to do that too or if not i'll just merge this.

My preference is to merge it now and then make further changes.

**Comment by dominichamon** [comment]
thank you!

**Pull request merged**